api/coordinatornetwork/libp2p.go

```diff
@@ -0,0 +1,212 @@
+package coordinatornetwork
+
+import (
+	"context"
+	"crypto/sha256"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/hermeznetwork/hermez-node/common"
+	"github.com/hermeznetwork/hermez-node/log"
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p"
+	connmgr "github.com/libp2p/go-libp2p-connmgr"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/routing"
+	discovery "github.com/libp2p/go-libp2p-discovery"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	yamux "github.com/libp2p/go-libp2p-yamux"
+	"github.com/libp2p/go-tcp-transport"
+	"github.com/mr-tron/base58"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/multiformats/go-multihash"
+)
+
+func setupHost(ctx context.Context, registeredCoordinators []common.Coordinator) (host.Host, *dht.IpfsDHT, error) {
```

Comment:
I am wondering where you plan to setup and register the coordinators slice. At cmd/main.go? 
Also, you need to be aware this list might be upgradable without need to restart hermeznode .  Maybe once in hour check if there is a new coordinator.

Reply:
> I am wondering where you plan to setup and register the coordinators slice. At cmd/main.go?

I believe this could end up being used outside of `hermez-node`, therefore I prefer to have a flexible approach in the sense of moving this logic into `node/node.go`. In our case I'm considering a few options:

a) Get the list from the DB, this is fast and easy but it has a problem: nodes that have just started for the first tie won't have coordinators stored on the DB (they have not synched anything yet)
b) Read the registered coordinators directly from the SC
c) Mix of the previous (if there are less than N coordinators on the DB, check it on Ethereum)

> Also, you need to be aware this list might be upgradable without need to restart hermeznode

Absolutely, the function `func (coordnet CoordinatorNetwork) FindMorePeers() error` should do the trick, although is hard to test this, if we find that this is not enough we could use the coordinator list from DB/SC again. In my experience while testing the connections were being handled nicely by using this function (nodes subscribed to the pubsub were appearing/disappearing correctly on each machine)

Reply:
> b) Read the registered coordinators directly from the SC
I believe this is most reliable option for starting.

>there is no need for all the nodes to be connected to all the nodes, from the official documentation
It makes sense. In general Ethereum nodes are connected to 5 other nodes. Six is a good number.

Reply:
> b) Read the registered coordinators directly from the SC
I believe this is most reliable option for starting.

>there is no need for all the nodes to be connected to all the nodes, from the official documentation
It makes sense. In general Ethereum nodes are connected to 5 other nodes. Six is a good number.